tests/test_optimizer.py

```diff
@@ -0,0 +1,135 @@
+import pytest
+import subprocess
+import shutil
+from utils import build_random_qasm, is_circuit_equiv
+from qiskit.qasm2 import load as load_qasm_2, LEGACY_CUSTOM_INSTRUCTIONS
+from pathlib import Path
+import time
+import random
+from wisq.guoq import GATE_SETS
+import csv
+
+def load_csv_args(path="args_limited.csv"):
+  rows = []
+  with open(path, newline="") as f:
+    reader = csv.DictReader(f)
+    for row in reader:
+      rows.append(row)
+  return rows
+
+
+def optimizer_cli_equivalence_test(
+  num_qubits: int = 4,
+  depth: int = 10,
+  quick: bool = False,
+  target_gateset: str | None = None,
+  csv_row: dict | None = None
```

Comment:
Maybe we can just have one parameter `args` that is a dictionary with everything that is currently in csv_row. And we can remove the `quick` param because if we want the test to be fast, we can just pass in a different CSV file.